homeassistant/components/leviosa_shades/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Leviosa shades Zone integration."""
+import logging
+
+from leviosapy import LeviosaZoneHub as tZoneHub
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.COVER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Leviosa shades Zone from a config entry."""
+    hub_name = entry.title
+    hub_ip = entry.data["host"]
+    hub = tZoneHub(
+        hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+    )
+    await hub.getHubInfo()  # Check all is good
+    _LOGGER.debug("Hub object created, FW: %s", hub.fwVer)
```

Comment:
Since you have details of your hub, you could add in a follow-up PR:

```python3
  device_registry.async_get_or_create(
            config_entry_id=entry.entry_id,
            identifiers={(DOMAIN, hub.mac)},
            manufacturer=MANUFACTURER,
            sw_version=hub.fwVer
        )
```

This will also nicely link the other entities to this hub, since you already use `via_device`. You can add a name/model to the hub as well, if you can detect this.

Reply:
OK I'll accepr your suggestion in next PR